{{cookiecutter.project_name}}/.devcontainer/postCreateCommand.sh

```diff
@@ -0,0 +1,19 @@
+#! /usr/bin/env bash
+
+# Install Dependencies
+poetry install --with dev
+
+# Check if the repo is already initialized
+if [ ! -d ".git" ]; then
```

Comment:
@kenibrewer Could you elaborate on why you added this? I prefer to keep this as simple as possible, and assume that people have initialized the Github repository as instructed in the `README.md`. For some the steps here might be considered/lead to unexpected behavior? However, if you think there is a good reason to keep this in, I am not opposed to it.

Reply:
@fpgmaas After you create a fresh child repo, the first thing you typically do is open the folder in VSCode and try to boot into the devcontainer. If the repo hasn't been initialized, the command `postCreate Command.sh` will fail when running `pre-commit install --install-hooks` because it requires an initialized git repo. I think it's better if a user's first experience with the devcontainer from this template isn't an error message. It's also better if the pre-commit hooks are actually installed right away instead of waiting until the second time you re-create you devcontainer. 

Beyond that, I added some additional code that will format everything using pre-commit right away. The intent was to prevent the problem you raised earlier about a github actions pipeline failing on an intial commit. We could get rid of those additional lines if you want to prioritize simplicity (which also makes sense to me).

Reply:
I think I'd prefer if we leave them out, since the user might not be aware this is happening in the background. In the situation where they first open the `README.md` and follow the steps, this should not be necessary. In the situation where they first open the devcontainer and then follow the `README.md` the steps there will give errors, since the devcontainer already took care of the steps such as `git init -b main`. This might lead to confusion.

Regarding the github actions pipeline failing on an iniial commit; I think it is great that we try to solve that to facilitate the implementation of the prettier pre-commit hooks and potentially others in the future. However, this might not be the right place since then we only solve said issue for people who use a devcontainer.

Do you agree with the above? Or maybe I am making a wrong assumption somewhere, I don't have prior experience with devcontainers.

Reply:
@fpgmaas I agree with your analysis. I'll remove the extra code.